pkg/condition/types/types_test.go

```diff
@@ -0,0 +1,142 @@
+package types
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/cel-go/cel"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrimitives(t *testing.T) {
```

Comment:
We should also have tests for conversions across types here. CEL interprets certain values across types.

e.g.

## Primitives
* string -> bool (should return error)
* number -> string (should return error)
* string -> float ("10" --> (float/double) 10 )
* float -> uint (10.0 --> uint64(10))
* float -> uint (10.1 --> return error, it cannot be represented by uint)
* negative float -> uint (-10.0 --> return error, it cannot be represented by uint)
* float -> int (10.0 --> int(10))
* string -> int ("10" --> int(10))

## Non-primitives
* invalid duration (e.g. "2sm")
* invalid timestamp (anything non-RFC3339, for example 2023-0914

Reply:
Related to this bit of code for numeric conversions:
https://github.com/openfga/openfga/pull/962/files#diff-613c5ff6a77df837179a9e734de3e33ec4fa5e219145710921d2266d8738f58dR27-R65